src/libs/actions/Welcome.js

```diff
@@ -36,7 +37,11 @@ Onyx.connect({
     key: ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER,
     initWithStoredValues: false,
     callback: (val) => {
-        isFirstTimeNewExpensifyUser = val;
+        // If this is a first time new expensify user, let's only update the state of isFirstTimeNewExpensifyUser from true to false
+        // after running all Welcome related logic in Welcome.show
+        if (!isFirstTimeNewExpensifyUser) {
```

Comment:
I'm a bit confused, you mention in the comment above this block updates isFirstTimeNewExpensifyUser from true to false. 

But for a new user, isFirstTimeNewExpensifyUser is first null from the top and then when backend returns true for `ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER` won't this be set from null to true?

Additionally from logic here if isFirstTimeNewExpensifyUser is true it will never set it to false. The condition here will give true only for `!false` so you can only change isFirstTimeNewExpensifyUser from false or null to true/false here.

So maybe your comment needs to be rephrased. Let me know if I'm mistaken.

Reply:
I think my comment is **very poorly written** @chiragsalian so I'll explain each question and rework the comment too.

> But for a new user, isFirstTimeNewExpensifyUser is first null from the top and then when backend returns true for ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER won't this be set from null to true?

Yes, the first time this logic is reached is when we first set `isFirstTimeNewExpensifyUser` to true.

> Additionally from logic here if isFirstTimeNewExpensifyUser is true it will never set it to false. The condition here will give true only for !false so you can only change isFirstTimeNewExpensifyUser from false or null to true/false here.

Yes! It only updates the value of `isFirstTimeNewExpensifyUser` if it already was falsy.
Since we never update `isFirstTimeNewExpensifyUser` to true, we only set `isFirstTimeNewExpensifyUser` when it first loads through Onyx.